syntax = "proto3";
// IMPORTANT to know: proto3 defines all fields as OPTIONAL BY DEFAULT!
// We make heavy usage of this in here!!!
import "google/protobuf/timestamp.proto";
package com.ptvgroup.archimedes.dataprv.protobuf.traffic.realtime.output.v1;
option java_package = "com.ptvgroup.archimedes.dataprv.protobuf.traffic.realtime.output.v1";
option java_outer_classname = "DataprvTrafficRealtimeData";
option java_multiple_files = false;
option csharp_namespace = "PTVGroup.Flows.Protos.DataPrv.Traffic.Realtime.Output.V1";

message DataprvTrafficRealtimeDataProto {
  string timezone = 1;
  google.protobuf.Timestamp snapshot_date_time = 2;
  repeated StreetTraffic street_traffic = 3;
  
  message StreetTraffic {
    reserved 4;
    int32 id = 1;
    int32 from_node_id = 2;
    double speed_kmh = 3;
    int32 probe_count = 5;  // Reserved for internal purposes and not reliable.
    string olr_code = 6;
  }
}
